kythe/proto/explore.proto

```diff
@@ -159,44 +157,37 @@ message TypeHierarchyReply {
 
 // Call graph
 // These APIs are for capturing the graphs induced by the _semantic_ relations between
-// functions (that is, the functions that call, or are called by, this function).
+// calling/callable nodes (for example, the functions that call, or are called by,
+// this function node).
 //
 // If you’re looking for a one-ply relationship between functions and the
 // _locations_ in the code at which they are called, Kythe XRefService's
 // CrossReferences API should do what you want.
 
-// Requests the functions that call the specified function.
+// Requests the incoming callgraphs for each of the specified nodes.
 message CallersRequest {
-  string function_ticket = 1;
-
-  NodeFilter node_filter = 2;
+  repeated string tickets = 1;
 }
 
 message CallersReply {
-  // same as function_ticket in request
-  string function_ticket = 1;
-
+  // Maps each input ticket to the corresponding call graph.
   // This is a graph, rather than a simple list of callers, to allow for
   // possible future expansion to recursive callers at depth > 1.
-  Graph graph = 2;
+  map<string, Graph> ticket_to_graph = 1;
```

Comment:
Do we care about shared graph structure?  Could we return a single Graph instead?

(Same for CalleesReply below)

Reply:
We could return a single Graph, yes.  My original thought was to preserve the connection between the input ticket and the output Graph, but on reflection that's probably not super important, and is overshadowed by the desirability of having a single Graph, so as to reduce the output size.